tm2/pkg/bft/config/toml.go

```diff
@@ -55,26 +55,30 @@ func WriteConfigFile(configFilePath string, config *Config) error {
 /****** these are for test settings ***********/
 
 func ResetTestRoot(testName string) *Config {
-	return ResetTestRootWithChainID(testName, "")
-}
+	chainID := "test-chain"
 
-func ResetTestRootWithChainID(testName string, chainID string) *Config {
 	// create a unique, concurrency-safe test directory under os.TempDir()
-	rootDir, err := os.MkdirTemp("", fmt.Sprintf("%s-%s_", chainID, testName))
+	testDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		panic(err)
+	}
+
+	rootDir, err := os.MkdirTemp(testDir, fmt.Sprintf("%s-%s_", chainID, testName))
 	if err != nil {
 		panic(err)
 	}
+
 	// ensure config and data subdirs are created
-	if err := osm.EnsureDir(filepath.Join(rootDir, defaultConfigDir), DefaultDirPerm); err != nil {
+	if err := osm.EnsureDir(filepath.Join(rootDir, defaultSecretsDir), DefaultDirPerm); err != nil {
 		panic(err)
 	}
-	if err := osm.EnsureDir(filepath.Join(rootDir, defaultDataDir), DefaultDirPerm); err != nil {
+	if err := osm.EnsureDir(filepath.Join(rootDir, DefaultDBDir), DefaultDirPerm); err != nil {
 		panic(err)
 	}
 
 	baseConfig := DefaultBaseConfig()
-	configFilePath := filepath.Join(rootDir, defaultConfigFilePath)
-	genesisFilePath := filepath.Join(rootDir, baseConfig.Genesis)
+	configFilePath := filepath.Join(rootDir, defaultConfigFileName)
+	genesisFilePath := filepath.Join(rootDir, "../", baseConfig.Genesis)
```

Comment:
While sorting out the tests for #1972, I realised that we probably shouldn't do this, and keep the config in the rootDir.

The reason is that the tests often use `os.RemoveAll` on the rootdir as a way to cleanup the directory. Furthermore, they will use the same genesis file, while they probably shouldn't.

So, let's revert this to the old one, adding a note.

Reply:
You're referring to keeping `genesis.json` by default in the node directory?

Reply:
>The reason is that the tests often use os.RemoveAll on the rootdir as a way to cleanup the directory. Furthermore, they will use the same genesis file, while they probably shouldn't.


I was waiting for someone to notice this change :)

I noticed a lot of the tests don't do proper cleanup with the temporary folders, and having the genesis outside by default requires a bit more setup for proper cleanup (I nested the folders in these helpers, because fixing them up was a propagation mess)

Reply:
Cherry picked the genesis changes here, to make this work:
[f59f4fa](https://github.com/gnolang/gno/pull/1944/commits/f59f4faea213140759cbada52d6390f8822c0326)